spec/models/run_spec.rb

```diff
@@ -118,33 +118,33 @@
     it 'has the correct splits' do
       expect(run.segments.map { |s| [s.segment_number, s.name, s.duration_ms(Run::REAL)] }).to match_array [
         [0, 'Tron City', 53_921],
-        [1, 'Start Abraxas fight', 185_453],
+        [1, 'Start Abraxas fight', 185_454],
         [2, 'Finish Abraxas fight', 108_446],
         [3, '"FLYNN!"', 151_874],
-        [4, '"Hey! Over here!"', 117_067],
-        [5, '"That was CLU. I saw him." (stack2 skip inc)', 44_819],
-        [6, '"Follow me."', 114_844],
+        [4, '"Hey! Over here!"', 117_068],
+        [5, '"That was CLU. I saw him." (stack2 skip inc)', 44_820],
+        [6, '"Follow me."', 114_845],
         [7, 'End of cycle ride (bridge skip inc)', 75_843],
-        [8, 'End of bridge thing', 431_475],
+        [8, 'End of bridge thing', 431_476],
         [9, 'Da Vinci', 18_882],
         [10, '"Hey! Over here!" #2', 418_688],
-        [11, 'Drive tank 4 feet', 372_653],
-        [12, 'Green guy', 283_038],
+        [11, 'Drive tank 4 feet', 372_654],
+        [12, 'Green guy', 283_039],
         [13, 'Beat games', 186_848],
-        [14, 'Green guy talks to me', 120_515],
-        [15, 'Get across that bridge', 484_387],
-        [16, 'Jump up and down on three lightsabers', 109_862],
+        [14, 'Green guy talks to me', 120_516],
+        [15, 'Get across that bridge', 484_388],
+        [16, 'Jump up and down on three lightsabers', 109_863],
         [17, '"That\'s Flynn\'s!"', 43_737],
-        [18, 'Race green guy', 67_095],
-        [19, 'Kill green guy', 220_725],
+        [18, 'Race green guy', 67_096],
+        [19, 'Kill green guy', 220_726],
         [20, 'Jump the bike', 278_102],
         [21, 'Grab me by the arm', 224_736],
-        [22, "Popcorn ceiling's revenge", 108_322],
+        [22, "Popcorn ceiling's revenge", 108_323],
         [23, 'Jump onto the flying staple', 112_977],
         [24, 'Ride motorbike for 9 seconds', 447_303],
-        [25, 'Final fight phase 1', 59_785],
-        [26, 'Final fight phase 2', 185_817],
-        [27, 'Final fight phase 3', 28_333]
+        [25, 'Final fight phase 1', 59_786],
+        [26, 'Final fight phase 2', 185_818],
+        [27, 'Final fight phase 3', 28_334]
```

Comment:
What caused the slight difference in durations here?

Reply:
Previous parser added all the segment times on top of each other using the run's total time.  At the end we use that sum to be the full timing for the run, but that was incorrect if you didn't actually finish the run making the last finished split before resetting the final time.
New parser takes each segment and calculates the durations using only the previous segment, then gets the final time from the last segments time.  This is why the open struct was used, to give us 0's for the first segment.

TL;DR: adding all segments vs just looking at each one compared to the last.